### ISSUE TITLE
machinectl continues even if pull-raw fails to download anything

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#14215]
```console
$ sudo machinectl pull-raw --verify=no \
                 https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.raw.xz Fedora-Cloud-Base-30-1.2.x86-64 
[sudo] password for zbyszek: 
Enqueued transfer job 1. Press C-c to continue download in background.
Pulling 'https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.raw.xz', saving as 'Fedora-Cloud-Base-30-1.2.x86-64'.
Downloading 0B for https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.nspawn.
Downloading 0B for https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.raw.xz.
Downloading 0B for https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.roothash.
Transfer failed: Couldn't resolve host name
Settings file could not be retrieved, proceeding without.
```

**Comment by poettering** [comment]
Hmm, any idea how I can reproduce this? What exactly was the error condition here?

I tried to reproduce this by using an URL that doesn't exist or by disconnecting the network, but it always handled these errors just fine.

**Comment by poettering** [comment]
Ah, I figured it out. Fix waiting in #14614
